src/dsp_tools/commands/ingest_xmlupload/upload_files/input_error.py

```diff
@@ -0,0 +1,69 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Protocol
+
+import pandas as pd
+
+separator = "\n\n"
+list_separator = "\n - "
+
+
+class Problem(Protocol):
+    """Information about input errors."""
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+        """
+
+
+@dataclass(frozen=True)
+class FileProblems(Problem):
+    """Handle the error communication to the user in case that some files don't exist or are unsupported."""
+
+    non_existing_files: list[Path]
+    unsupported_files: list[Path]
+    maximum_prints: int = 50
+
+    def __post_init__(self) -> None:
+        if not self.non_existing_files and not self.unsupported_files:
+            raise ValueError("It's not possible to create a FileProblems object without any problems.")
+
+    def execute_error_protocol(self) -> str:
+        """
+        Generate the error message to communicate the problems to the user.
+        If there are too many problems, save them to a file.
+
+        Returns:
+            error message
+        """
+        msg = "Some files referenced in the <bitstream> tags of your XML file cannot be uploaded to the server."
+        if len(self.non_existing_files) + len(self.unsupported_files) > self.maximum_prints:
+            output_file = Path("file_problems.csv")
+            self._save_to_csv(output_file)
+            msg += f" The full list of files with problems has been saved to '{output_file}'."
+            return msg
+        if self.non_existing_files:
+            msg += separator
+            msg += "The following files don't exist on your computer:"
+            msg += list_separator + list_separator.join([str(file) for file in self.non_existing_files])
+        if self.unsupported_files:
+            msg += separator
+            msg += "The following files have unsupported extensions:"
+            msg += list_separator + list_separator.join([str(file) for file in self.unsupported_files])
+        return msg
+
+    def _save_to_csv(self, output_file: Path) -> None:
+        non_existing, unsupported = self._add_padding()
+        data = {
+            "Files that don't exist on your computer": non_existing,
+            "Files with unsupported extensions": unsupported,
+        }
+        df = pd.DataFrame(data)
+        df.to_csv(output_file, index=False)
+
+    def _add_padding(self) -> tuple[list[Path | None], list[Path | None]]:
+        max_len = max(len(self.non_existing_files), len(self.unsupported_files))
+        non_existing = self.non_existing_files + [None] * (max_len - len(self.non_existing_files))
+        unsupported = self.unsupported_files + [None] * (max_len - len(self.unsupported_files))
+        return non_existing, unsupported
```

Comment:
due to the changes that were pushed while I was reviewing the code, I'm not sure if my comments on these will still be showing up... if so, sorry for the duplication.  
in short: I don't like the structure if this report, it's against the tabular logic of "one line per thing".

Also, what's the deal with the padding? If that is formatting the CSV, then that's against the logic of structured data in general.